### PR TITLE
Drop build postgresql 9.6

### LINKED ISSUE #16303: Drop postgresql 9.6 in Harbor 2.7
Whether to remove postgresql 9.6 in Harbor 2.7？

### PR DESCRIPTION
**Pull request by YangJiao0817**
Drop pg9.6, will not support upgrading from pg9.6 to pg13, if you use pg9.6 you can upgrade to 2.5 and upgrade to 2.7

Signed-off-by: Yang Jiao <jiaoya@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #16303

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: make/photon/db/docker-entrypoint.sh [hunk 1]
```diff
@@ -11,50 +11,38 @@ PGBINOLD="/usr/local/pg${PG_VERSION_OLD}/bin"
...
-                set -e
-                echo "remove the $PGDATAOLD after upgrade success."
-                if [ "$PGDATAOLD" = "$PGDATA" ]; then
-                        find $PGDATA/* -prune ! -name pg${PG_VERSION_NEW} -exec rm -rf {} \;
-                else
-                        rm -rf $PGDATAOLD
-                fi
-        else
-                echo "no need to upgrade postgres, launch it."
+PGDATEDIR=$([ $PG_VERSION_NEW ] && echo $PGDATANEW || echo $PGDATAOLD)
```
Comment (wy65701436) [inline make/photon/db/docker-entrypoint.sh:14]: you should update the PGBINOLD

Comment (wy65701436) [inline make/photon/db/docker-entrypoint.sh:14]: please use a func to get PGDATEDIR, rathen than the complicated logic statement.

File: make/photon/db/docker-entrypoint.sh [hunk 2]
```diff
@@ -11,50 +11,38 @@ PGBINOLD="/usr/local/pg${PG_VERSION_OLD}/bin"
...
+
+# Upgrade DB: 1. has $PG_VERSION_NEW. 2. without pg new version PG_VERSION. 3. has pg old version PG_VERSION.
+# Init DB:    1. has $PG_VERSION_NEW. 2. without pg new version PG_VERSION. 3. without pg old version PG_VERSION.
+#             1. without $PG_VERSION_NEW. 2. without pg old version PG_VERSION.
+if [ $PG_VERSION_NEW ] && [ ! -s $PGDATANEW/PG_VERSION ] && [ -s $PGDATAOLD/PG_VERSION ]; then
```
Comment (wy65701436) [inline make/photon/db/docker-entrypoint.sh:19]: if -z $PG_VERSION_NEW?

File: make/photon/db/docker-entrypoint.sh [hunk 3]
```diff
@@ -11,50 +11,38 @@ PGBINOLD="/usr/local/pg${PG_VERSION_OLD}/bin"
...
-                echo "remove the $PGDATAOLD after upgrade success."
-                if [ "$PGDATAOLD" = "$PGDATA" ]; then
-                        find $PGDATA/* -prune ! -name pg${PG_VERSION_NEW} -exec rm -rf {} \;
-                else
-                        rm -rf $PGDATAOLD
-                fi
-        else
-                echo "no need to upgrade postgres, launch it."
+PGDATEDIR=$([ $PG_VERSION_NEW ] && echo $PGDATANEW || echo $PGDATAOLD)
+
```
Comment (wy65701436) [inline make/photon/db/docker-entrypoint.sh:15]:
```
# we should block the upgrade path from 9.6 directly.
if [ -s $PGDATA/PG_VERSION ]; then
          echo "$error msg"
          exit 1
fi
```

and it needs test cases to cover the below patches
- 2.2 - 2.7
- 2.2 - 2.4/2.5 - 2.7

**Review by stonezdj — APPROVED** [review]
LGTM

**Comment by stonezdj** [comment]
We'd verify if the helm upgrade fail because of the pg96 database, then the helm rollback command could work and roll back to the previous workable release.

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**